File: frontend/src/types/usage.ts
import type { DecimalString } from "./chat";

export interface UsageSummary {
  user_id: string;
  total_credits: DecimalString;
  credits_used: DecimalString;
  credits_remaining: DecimalString;
  request_count: number;
  total_input_tokens: number;
  total_output_tokens: number;
  total_tokens: number;
  total_cost: DecimalString;
  currency: string;
}

File: backend/app/schemas/usage.py
from pydantic import BaseModel

from app.schemas.common import DecimalStr


class UsageSummaryResponse(BaseModel):
    user_id: str
    total_credits: DecimalStr
    credits_used: DecimalStr
    credits_remaining: DecimalStr
    request_count: int
    total_input_tokens: int
    total_output_tokens: int
    total_tokens: int
    total_cost: DecimalStr
    currency: str = "USD"
